src/get-started/flutter-for/ios-devs.md

```diff
@@ -5,1029 +5,1484 @@ description: Learn how to apply iOS developer knowledge when building Flutter ap
+</ul>
+</nav>
+
+<?code-excerpt "lib/list.dart (SimpleList)"?>
+```dart
+class Person {
+  String name;
+  Person(this.name);
+}
 
-    return CupertinoButton(
-      onPressed: () {},
-      child: const Text('Toggle Two'),
-    );
-  }
+var items = [
+  Person('Person 1'),
+  Person('Person 2'),
+  Person('Person 3'),
+];
+
+class HomePage extends StatelessWidget {
+  const HomePage({super.key});
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      appBar: AppBar(
-        title: const Text('Sample App'),
-      ),
-      body: Center(
-        child: _getToggleChild(),
-      ),
-      floatingActionButton: FloatingActionButton(
-        onPressed: _toggle,
-        tooltip: 'Update Text',
-        child: const Icon(Icons.update),
+      body: ListView.builder(
+        itemCount: items.length,
+        itemBuilder: (context, index) {
+          return ListTile(
+            title: Text(items[index].name),
+          );
+        },
       ),
     );
   }
 }
 ```
 
-### How do I animate a widget?
-
-In iOS, you create an animation by calling the
-`animate(withDuration:animations:)` method on a view.
-In Flutter, use the animation library
-to wrap widgets inside an animated widget.
+Flutter has some caveats for lists:
 
-In Flutter, use an `AnimationController`, which is an `Animation<double>`
-that can pause, seek, stop, and reverse the animation.
-It requires a `Ticker` that signals when vsync happens
-and produces a linear interpolation
-between 0 and 1 on each frame while it's running.
-You then create one or more
-`Animation`s and attach them to the controller.
+- The [`ListView`] widget has a builder method.
+This works like the `ForEach` within SwiftUI's `List` struct.
 
-For example, you might use `CurvedAnimation`
-to implement an animation along an interpolated curve.
-In this sense, the controller is the "master" source
-of the animation progress
-and the `CurvedAnimation` computes the curve
-that replaces the controller's default linear motion.
-Like widgets, animations in Flutter work with composition.
+- The `itemCount` parameter of the `ListView` sets how many items
+that the `ListView` displays.
 
-When building the widget tree you assign the `Animation` to an animated
-property of a widget, such as the opacity of a `FadeTransition`,
-and tell the controller to start the animation.
+- The `.builder` method calls the `itemBuilder` widget with bounds.
+  - Set the lower bound to any positive number including 0.
+  - Set the upper bound to any positive number excluding the item count.
+  - This widget must return one `Widget` instance per list item.
 
-The following example shows how to write a `FadeTransition` that
-fades the widget into a logo when you press the `FloatingActionButton`:
+The previous example returned a [`ListTile`][] widget for each item.
+The `ListTile` widget includes properties like `height` and `font-size`.
+These properties help build a list. Flutter allows you to return
```

Comment:
Adding "However" to make it clear that even though ListTile is useful for widgets, you still can use other widgets. That point wasn't clear to me without this word